### PR TITLE
[codex] Populate young IPO partial metrics

### PR DESCRIPTION
**Pull request by xang1234**
## What changed

- Populates short-history metrics for young IPO scan rows when enough data exists: RS/price sparklines, 1-day price change, ADR%, and 1M RS.
- Keeps long-history-only fields blank for young rows and preserves nullable composite-score semantics.
- Persists visible `insufficient_history` rows in dynamic/API scan results so they appear in result tables, search, and CSV exports.
- Adds regression coverage across orchestrator behavior, dynamic scan persistence, CSV export, feature-store/static serialization, and frontend table rendering.

## Why this changed

PR #102 made young IPOs visible in static scanner results, but dynamic/API-backed scan paths could still drop or under-populate those rows. Some unavailable metrics also risked being represented as `0`/`NaN` instead of `null`.

## User impact

Young IPOs remain visible in scanner results while showing calculable short-history values where available. Missing long-history values remain blank instead of looking like real zero scores.

## Validation

- `cd backend && source venv/bin/activate && DATABASE_URL=sqlite:// STOCKSCANNER_TEST_ALLOW_SQLITE=1 pytest tests/unit/test_scan_orchestrator.py tests/unit/use_cases/test_run_bulk_scan.py tests/unit/use_cases/test_export_scan_results.py tests/unit/use_cases/feature_store/test_build_daily_snapshot.py tests/unit/test_static_site_export_service.py tests/integration/test_feature_store_scan_results.py` -> 130 passed
- `cd frontend && npm run test -- src/components/Scan/ResultsTable.test.jsx src/static/scanClient.test.js` -> 35 passed
- `cd frontend && npm run lint` -> 0 errors, 1 pre-existing warning in `frontend/src/static/StaticMarketContext.jsx`

## Summary by CodeRabbit

* **New Features**
  * Added partial metrics computation for stocks with insufficient history (e.g., young IPOs), including price change, ADR%, and relative strength ratings where available.

* **Bug Fixes**
  * Improved persistence of scan results with insufficient history instead of skipping them.
  * Enhanced null and edge case handling in metric calculations.

* **Tests**
  * Expanded test coverage for insufficient-history scenarios and young IPO handling.

### CODE REVIEW

File: backend/app/scanners/scan_orchestrator.py [hunk 1]
```diff
@@ -173,6 +176,126 @@ def _scan_mode_for_history(history_bars: int) -> str:
...
+                if stock_data.rs_universe_performances
+                else None,
+            )
+
+    if price_data is not None and not price_data.empty:
+        metrics["adr_percent"] = ADRCalculator().calculate_adr_percent(
+            price_data,
+            period=20,
+            min_valid_rows=20,
+        )
```
[the remark this answers is not in the document]
Comment (xang1234) [reply backend/app/scanners/scan_orchestrator.py:296]: Addressed in `backend/app/scanners/scan_orchestrator.py`: added a comment documenting that young IPO partial metrics intentionally require a fully valid 20-session ADR window instead of the scanner-wide 80% tolerance. Verified with `pytest tests/unit/test_scan_orchestrator.py::TestScanOrchestratorErrorPaths::test_young_ipo_adr_requires_twenty_valid_ohlc_rows -q` and `python -m py_compile app/scanners/scan_orchestrator.py`.
